### DISCUSSION TITLE
Maybe we should Show the test coverage of the current

### DISCUSSION THREAD

**Discussion by zhuxiujia** [vlang/v#18600] [Ideas]
Maybe we should improve the test coverage of v

and  Show the test coverage of the current v   just like    tested =  8% 

![tested](https://s3.amazonaws.com/assets.coveralls.io/badges/coveralls_23.svg)

I think that nearly 90% of the test coverage will be as stable as sqlite, with few bugs.

Even for the v standard library, untested code is not allowed to be released！！！（Maybe it should be called v-test-scanner）

**Comment by ghost** [comment]
Having a coverage report would be very valuable to me. In my organization, we keep our repositories in Bitbucket and hook them into SonarCloud, and on every PR, run SonarCloud analysis and prevent merges if SonarCloud QualityGate reports any rule violations. One of our internal rules is a certain level of test coverage, which SonarCloud determines by analyzing a coverage report. I don't know if SonarCloud would understand a Vlang coverage report, but there may be a "standard" format that could be used. If not, it's at least a place to start.
